runtime/src/main/kotlin/kotlin/native/internal/KSuspendFunctionImpl.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.internal
+
+import kotlin.reflect.KType
+import kotlin.reflect.KSuspendFunction
+
+@FixmeReflection
+internal abstract class KSuspendFunctionImpl<out R>(
+        override val name: String, val fqName: String,
+        val bound: Boolean, val receiver: Any?,
+        override val returnType: KType
+): KSuspendFunction<R> {
+
+    override fun equals(other: Any?): Boolean {
+        if (other !is KSuspendFunctionImpl<*>) return false
+        return fqName == other.fqName && bound == other.bound && receiver == other.receiver
```

Comment:
Can they differ by number or type of parameters?

Reply:
Hmm, they might actually be different by number of parameters. This was copied from `KFunctionImpl`, so there is a bug there as well.

Reply:
As it turns out, `fqName` here is our internal `symbolName`, so everything works fine. Still, adding `arity` instead of `bound` won't hurt and will bring us closer to Kotlin/JVM.

Reply:
But may be do it in a separate PR? It's not connected with the purpose of this PR.